File: scripts/war-briefing.js
#!/usr/bin/env node
/**
 * 美以伊战争自动简报脚本
 * 每天4个时间点执行，汇总前6小时情况
 */

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

// 时间配置
const now = new Date();
const beijingTime = new Date(now.getTime() + 8 * 60 * 60 * 1000); // UTC+8
const timeSlot = getTimeSlot(beijingTime.getHours());
const period = getPeriod(timeSlot);

console.log(`[${now.toISOString()}] 开始生成战争简报...`);
console.log(`时间点: ${timeSlot}:00 (北京时间)`);
console.log(`汇总时段: ${period}`);

// 搜索最新战争新闻
async function generateBriefing() {
    try {
        // 使用tavily搜索最新消息
        const searchQuery = `美国 以色列 伊朗 战争 最新消息 ${beijingTime.getFullYear()}年${beijingTime.getMonth() + 1}月${beijingTime.getDate()}日`;
        
        console.log(`搜索关键词: ${searchQuery}`);
        
        const searchResult = execSync(
            `cd /home/zhangyufeng/.openclaw/workspace && python3 ./skills/openclaw-tavily-search/scripts/tavily_search.py --query "${searchQuery}" --max-results 10 --format brave 2>&1`,
            { encoding: 'utf-8', maxBuffer: 10 * 1024 * 1024 }
        );
        
        let searchData;
        try {
            searchData = JSON.parse(searchResult);
        } catch (e) {
            console.error('解析搜索结果失败:', e.message);
            searchData = { results: [] };
        }
        
        // 生成简报
        const briefing = generateBriefingContent(searchData.results || [], timeSlot, period);
        
        // 保存简报文件
        const briefingDir = '/home/zhangyufeng/.openclaw/workspace/briefings';
        if (!fs.existsSync(briefingDir)) {
            fs.mkdirSync(briefingDir, { recursive: true });
        }
        
        const filename = `war-briefing-${beijingTime.toISOString().split('T')[0]}-${timeSlot}.md`;
        const filepath = path.join(briefingDir, filename);
        
        fs.writeFileSync(filepath, briefing);
        console.log(`简报已保存: ${filepath}`);
        
        // 发送到Telegram
        await sendToTelegram(briefing);
        
        return briefing;
        
    } catch (error) {
        console.error(`生成简报失败:`, error.message);
        // 发送错误通知
        await sendToTelegram(`⚠️ 战争简报生成失败 (${timeSlot}:00)\n错误: ${error.message}`);
        throw error;
    }
}

// 生成简报内容
function generateBriefingContent(results, timeSlot, period) {
    const timestamp = beijingTime.toLocaleString('zh-CN', { 
        timeZone: 'Asia/Shanghai',
        year: 'numeric',
        month: '2-digit',
        day: '2-digit',
        hour: '2-digit',
        minute: '2-digit',
        second: '2-digit'
    });
    
    let briefing = `## 📊 美以伊战争简报 #${timeSlot/6}\n`;
    briefing += `**时间点**: ${timeSlot}:00 (北京时间)\n`;
    briefing += `**汇总时段**: ${period}\n`;
    briefing += `**生成时间**: ${timestamp}\n`;
    briefing += `---\n\n`;
    
    if (results.length === 0) {
        briefing += `### ⚠️ 无最新消息\n`;
        briefing += `过去6小时内未发现重大战况更新。\n`;
        briefing += `可能原因：\n`;
        briefing += `1. 战况相对稳定\n`;
        briefing += `2. 信息延迟\n`;
        briefing += `3. 搜索API限制\n\n`;
        briefing += `**建议**: 手动检查权威新闻源。\n`;
        return briefing;
    }
    
    // 分析搜索结果
    const casualties = extractCasualties(results);
    const keyEvents = extractKeyEvents(results);
    const politicalUpdates = extractPoliticalUpdates(results);
    const regionalImpact = extractRegionalImpact(results);
    
    // 1. 核心战况
    briefing += `### 一、核心战况更新\n\n`;
    if (casualties) {
        briefing += `**伤亡统计**: ${casualties}\n\n`;
    }
    if (keyEvents.length > 0) {
        briefing += `**关键事件**:\n`;
        keyEvents.forEach(event => {
            briefing += `- ${event}\n`;
        });
        briefing += `\n`;
    }
    
    // 2. 军事行动
    briefing += `### 二、军事行动\n\n`;
    const militaryActions = extractMilitaryActions(results);
    if (militaryActions.length > 0) {
        militaryActions.forEach(action => {
            briefing += `- ${action}\n`;
        });
    } else {
        briefing += `- 无重大军事行动更新\n`;
    }
    briefing += `\n`;
    
    // 3. 政治动态
    briefing += `### 三、政治动态\n\n`;
    if (politicalUpdates.length > 0) {
        politicalUpdates.forEach(update => {
            briefing += `- ${update}\n`;
        });
    } else {
        briefing += `- 无重大政治动态更新\n`;
    }
    briefing += `\n`;
    
    // 4. 地区影响
    briefing += `### 四、地区影响\n\n`;
    if (regionalImpact.length > 0) {
        regionalImpact.forEach(impact => {
            briefing += `- ${impact}\n`;
        });
    } else {
        briefing += `- 无新增地区影响\n`;
    }
    briefing += `\n`;
    
    // 5. 简报评价
    briefing += `### 五、简报评价\n\n`;
    briefing += `**局势评级**: ${rateSituation(results)}\n`;
    briefing += `**冲突级别**: ${rateConflictLevel(results)}\n`;
    briefing += `**扩散风险**: ${rateSpreadRisk(results)}\n\n`;
    
    // 6. 下一份简报
    briefing += `### 六、下一份简报\n\n`;
    briefing += `**时间**: ${(timeSlot + 6) % 24}:00 (北京时间)\n`;
    briefing += `**汇总时段**: ${getPeriod((timeSlot + 6) % 24)}\n\n`;
    
    // 7. 信息来源
    briefing += `### 七、信息来源\n\n`;
    results.slice(0, 3).forEach((result, index) => {
        briefing += `${index + 1}. [${result.title}](${result.url})\n`;
    });
    
    return briefing;
}

// 辅助函数
function getTimeSlot(hour) {
    if (hour >= 0 && hour < 6) return 0;
    if (hour >= 6 && hour < 12) return 6;
    if (hour >= 12 && hour < 18) return 12;
    return 18;
}

function getPeriod(timeSlot) {
    const periods = {
        0: '前日18:00-今日00:00',
        6: '00:00-06:00',
        12: '06:00-12:00',
        18: '12:00-18:00'
    };
    return periods[timeSlot] || '未知时段';
}

function extractCasualties(results) {
    for (const result of results) {
        if (result.snippet && result.snippet.includes('死亡') && result.snippet.includes('受伤')) {
            const match = result.snippet.match(/(\d+)\s*死亡[，、]\s*(\d+)\s*受伤/);
            if (match) {
                return `${match[1]}人死亡，${match[2]}人受伤`;
            }
        }
    }
    return null;
}

function extractKeyEvents(results) {
    const events = [];
    const keywords = ['小学', '医院', '平民', '儿童', '学校', '居民区'];
    
    for (const result of results) {
        if (result.snippet) {
            for (const keyword of keywords) {
                if (result.snippet.includes(keyword) && !events.some(e => e.includes(keyword))) {
                    events.push(result.snippet.substring(0, 100) + '...');
                    break;
                }
            }
        }
    }
    
    return events.slice(0, 3);
}

function extractPoliticalUpdates(results) {
    const updates = [];
    const keywords = ['特朗普', '哈梅内伊', '以色列总理', '联合国', '国会', '谈判'];
    
    for (const result of results) {
        if (result.snippet) {
            for (const keyword of keywords) {
                if (result.snippet.includes(keyword) && !updates.some(u => u.includes(keyword))) {
                    updates.push(result.snippet.substring(0, 100) + '...');
                    break;
                }
            }
        }
    }
    
    return updates.slice(0, 3);
}

function extractRegionalImpact(results) {
    const impacts = [];
    const regions = ['卡塔尔', '阿联酋', '科威特', '沙特', '巴林', '约旦'];
    
    for (const result of results) {
        if (result.snippet) {
            for (const region of regions) {
                if (result.snippet.includes(region) && !impacts.some(i => i.includes(region))) {
                    impacts.push(`${region}: ${result.snippet.substring(0, 80)}...`);
                    break;
                }
            }
        }
    }
    
    return impacts.slice(0, 3);
}

function extractMilitaryActions(results) {
    const actions = [];
    const keywords = ['袭击', '打击', '导弹', '无人机', '航母', '基地'];
    
    for (const result of results) {
        if (result.snippet) {
            const snippet = result.snippet;
            for (const keyword of keywords) {
                if (snippet.includes(keyword)) {
                    // 提取包含关键词的句子
                    const sentences = snippet.split(/[。.!?]/);
                    for (const sentence of sentences) {
                        if (sentence.includes(keyword) && sentence.length > 20) {
                            actions.push(sentence.trim());
                            break;
                        }
                    }
                    break;
                }
            }
        }
    }
    
    return [...new Set(actions)].slice(0, 5);
}

function rateSituation(results) {
    if (results.length === 0) return '🟡 待观察';
    
    const criticalKeywords = ['死亡', '袭击', '爆炸', '打击', '导弹'];
    let criticalCount = 0;
    
    for (const result of results) {
        if (result.snippet) {
            for (const keyword of criticalKeywords) {
                if (result.snippet.includes(keyword)) {
                    criticalCount++;
                    break;
                }
            }
        }
    }
    
    if (criticalCount >= 5) return '🔥🔥🔥🔥 高度危险';
    if (criticalCount >= 3) return '🔥🔥🔥 危险升级';
    if (criticalCount >= 1) return '🔥🔥 紧张持续';
    return '🔥 相对稳定';
}

function rateConflictLevel(results) {
    const keywords = ['全面战争', '大规模', '持续', '有限', '局部'];
    
    for (const result of results) {
        if (result.snippet) {
            if (result.snippet.includes('全面战争') || result.snippet.includes('大规模')) {
                return '全面军事对抗';
            }
            if (result.snippet.includes('持续')) {
                return '持续冲突';
            }
            if (result.snippet.includes('有限') || result.snippet.includes('局部')) {
                return '有限冲突';
            }
        }
    }
    
    return '待评估';
}

function rateSpreadRisk(results) {
    const regionKeywords = ['卡塔尔', '阿联酋', '科威特', '沙特', '巴林', '约旦', '黎巴嫩', '叙利亚'];
    let regionCount = 0;
    
    for (const result of results) {
        if (result.snippet) {
            for (const keyword of regionKeywords) {
                if (result.snippet.includes(keyword)) {
                    regionCount++;
                    break;
                }
            }
        }
    }
    
    if (regionCount >= 3) return '高';
    if (regionCount >= 1) return '中';
    return '低';
}

// 发送到Telegram
async function sendToTelegram(briefing) {
    try {
        // 截取前2000字符（Telegram消息限制）
        const message = briefing.length > 2000 ? briefing.substring(0, 2000) + '...\n\n(完整简报已保存)' : briefing;
        
        // 使用OpenClaw的message工具发送
        const messageTool = require('/home/zhangyufeng/.npm-global/lib/node_modules/openclaw/dist/tools/message.js');
        
        // 这里需要实际的发送逻辑
        // 由于工具调用复杂，先保存到文件，后续通过cron job发送
        
        console.log(`简报已生成，长度: ${briefing.length}字符`);
        console.log(`消息预览: ${message.substring(0, 100)}...`);
        
        return true;
        
    } catch (error) {
        console.error('发送到Telegram失败:', error.message);
        return false;
    }
}

// 执行
generateBriefing().then(() => {
    console.log(`[${now.toISOString()}] 战争简报生成完成`);
    process.exit(0);
}).catch(error => {
    console.error(`[${now.toISOString()}] 战争简报生成失败:`, error);
    process.exit(1);
});

File: scripts/war-briefing.go
// war-briefing.go
// 美以伊战争自动简报脚本（Go版本）
// 每天4个时间点执行，汇总前6小时情况

package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"os"
	"os/exec"
	"path/filepath"
	"strings"
	"time"
)

// SearchResult 搜索结果
type SearchResult struct {
	Title   string `json:"title"`
	URL     string `json:"url"`
	Snippet string `json:"snippet"`
}

// SearchResponse 搜索响应
type SearchResponse struct {
	Query   string         `json:"query"`
	Results []SearchResult `json:"results"`
}

// Briefing 简报结构
type Briefing struct {
	TimeSlot    string   `json:"time_slot"`
	Period      string   `json:"period"`
	GeneratedAt string   `json:"generated_at"`
	Content     string   `json:"content"`
	Sources     []string `json:"sources"`
}

// 常量
const (
	workspaceDir = "/home/zhangyufeng/.openclaw/workspace"
	briefingDir  = "/home/zhangyufeng/.openclaw/workspace/briefings"
)

func main() {
	log.Println("📊 战争简报脚本启动（Go版本）")

	// 获取当前时间（北京时间）
	now := time.Now()
	beijingTime := now.Add(8 * time.Hour) // UTC+8
	hour := beijingTime.Hour()

	// 确定时间点和时段
	timeSlot, period := getTimeSlot(hour)
	
	log.Printf("时间点: %02d:00 (北京时间)", timeSlot)
	log.Printf("汇总时段: %s", period)

	// 生成简报
	briefing, err := generateBriefing(timeSlot, period, beijingTime)
	if err != nil {
		log.Fatalf("❌ 生成简报失败: %v", err)
	}

	// 保存简报
	if err := saveBriefing(briefing, beijingTime); err != nil {
		log.Fatalf("❌ 保存简报失败: %v", err)
	}

	// 输出简报内容
	fmt.Println("\n" + briefing.Content)

	log.Println("✅ 简报生成完成")
}

// getTimeSlot 获取时间点和时段
func getTimeSlot(hour int) (int, string) {
	var timeSlot int
	var period string

	switch {
	case hour >= 0 && hour < 6:
		timeSlot = 0
		period = "前日18:00-今日00:00"
	case hour >= 6 && hour < 12:
		timeSlot = 6
		period = "00:00-06:00"
	case hour >= 12 && hour < 18:
		timeSlot = 12
		period = "06:00-12:00"
	default:
		timeSlot = 18
		period = "12:00-18:00"
	}

	return timeSlot, period
}

// generateBriefing 生成简报
func generateBriefing(timeSlot int, period string, beijingTime time.Time) (*Briefing, error) {
	// 搜索最新消息
	results, err := searchWarNews(beijingTime)
	if err != nil {
		return nil, fmt.Errorf("搜索失败: %v", err)
	}

	// 生成简报内容
	content := generateContent(results, timeSlot, period, beijingTime)

	// 提取信息来源
	sources := extractSources(results)

	return &Briefing{
		TimeSlot:    fmt.Sprintf("%02d:00", timeSlot),
		Period:      period,
		GeneratedAt: beijingTime.Format("2006-01-02 15:04:05"),
		Content:     content,
		Sources:     sources,
	}, nil
}

// searchWarNews 搜索战争新闻
func searchWarNews(beijingTime time.Time) ([]SearchResult, error) {
	// 构建搜索查询
	query := fmt.Sprintf("美国 以色列 伊朗 战争 最新消息 %d年%d月%d日",
		beijingTime.Year(),
		beijingTime.Month(),
		beijingTime.Day())

	log.Printf("搜索查询: %s", query)

	// 调用tavily-search技能
	cmd := exec.Command("python3", 
		filepath.Join(workspaceDir, "skills/openclaw-tavily-search/scripts/tavily_search.py"),
		"--query", query,
		"--max-results", "8",
		"--format", "brave")

	cmd.Dir = workspaceDir
	output, err := cmd.CombinedOutput()
	if err != nil {
		log.Printf("⚠️ 搜索命令执行错误: %v", err)
		log.Printf("输出: %s", output)
		return []SearchResult{}, nil // 返回空结果，不中断流程
	}

	// 解析JSON响应
	var response SearchResponse
	if err := json.Unmarshal(output, &response); err != nil {
		log.Printf("⚠️ 解析搜索结果失败: %v", err)
		log.Printf("原始输出: %s", string(output[:min(200, len(output))]))
		return []SearchResult{}, nil
	}

	log.Printf("找到 %d 条结果", len(response.Results))
	return response.Results, nil
}

// generateContent 生成简报内容
func generateContent(results []SearchResult, timeSlot int, period string, beijingTime time.Time) string {
	var content strings.Builder

	// 标题
	content.WriteString(fmt.Sprintf("📊 *美以伊战争简报* (#%d)\n\n", timeSlot/6+1))
	content.WriteString(fmt.Sprintf("*时间*: %02d:00 (北京时间)\n", timeSlot))
	content.WriteString(fmt.Sprintf("*时段*: %s\n", period))
	content.WriteString(fmt.Sprintf("*生成*: %s\n\n", 
		beijingTime.Format("2006/01/02 15:04")))

	if len(results) == 0 {
		content.WriteString("⚠️ *无最新消息*\n")
		content.WriteString("过去6小时未发现重大战况更新。\n")
		content.WriteString("可能原因：\n")
		content.WriteString("1. 战况相对稳定\n")
		content.WriteString("2. 信息延迟\n")
		content.WriteString("3. 搜索API限制\n\n")
	} else {
		// 提取关键信息
		casualties := extractCasualties(results)
		keyEvents := extractKeyEvents(results, 3)
		politicalUpdates := extractPoliticalUpdates(results, 2)
		militaryActions := extractMilitaryActions(results, 3)

		// 伤亡统计
		if casualties != "" {
			content.WriteString(fmt.Sprintf("💀 *伤亡*: %s\n", casualties))
		}

		// 关键事件
		if len(keyEvents) > 0 {
			content.WriteString("\n📰 *关键事件*:\n")
			for _, event := range keyEvents {
				content.WriteString(fmt.Sprintf("• %s\n", event))
			}
		}

		// 政治动态
		if len(politicalUpdates) > 0 {
			content.WriteString("\n🏛️ *政治动态*:\n")
			for _, update := range politicalUpdates {
				content.WriteString(fmt.Sprintf("• %s\n", update))
			}
		}

		// 军事行动
		if len(militaryActions) > 0 {
			content.WriteString("\n⚔️ *军事行动*:\n")
			for _, action := range militaryActions {
				content.WriteString(fmt.Sprintf("• %s\n", action))
			}
		}

		content.WriteString("\n")
	}

	// 局势评估
	content.WriteString("*评估*:\n")
	rating := assessSituation(results)
	content.WriteString(fmt.Sprintf("%s\n\n", rating))

	// 下一简报
	nextSlot := (timeSlot + 6) % 24
	content.WriteString(fmt.Sprintf("*下一简报*: %02d:00\n", nextSlot))

	return content.String()
}

// extractCasualties 提取伤亡信息
func extractCasualties(results []SearchResult) string {
	for _, result := range results {
		snippet := result.Snippet
		if strings.Contains(snippet, "死亡") && strings.Contains(snippet, "受伤") {
			// 尝试提取数字
			lines := strings.Split(snippet, "。")
			for _, line := range lines {
				if strings.Contains(line, "死亡") && strings.Contains(line, "受伤") {
					// 简化处理
					return extractNumbers(line)
				}
			}
		}
	}
	return ""
}

// extractKeyEvents 提取关键事件
func extractKeyEvents(results []SearchResult, limit int) []string {
	var events []string
	keywords := []string{"小学", "医院", "平民", "儿童", "学校", "居民区", "爆炸", "袭击"}

	for _, result := range results {
		if len(events) >= limit {
			break
		}
		
		title := result.Title
		snippet := result.Snippet
		
		// 检查是否包含关键词
		for _, keyword := range keywords {
			if strings.Contains(title, keyword) || strings.Contains(snippet, keyword) {
				// 简化标题
				event := simplifyText(title, 60)
				if event != "" {
					events = append(events, event)
					break
				}
			}
		}
	}
	
	return events
}

// extractPoliticalUpdates 提取政治动态
func extractPoliticalUpdates(results []SearchResult, limit int) []string {
	var updates []string
	keywords := []string{"特朗普", "哈梅内伊", "以色列总理", "联合国", "国会", "谈判"}

	for _, result := range results {
		if len(updates) >= limit {
			break
		}
		
		snippet := result.Snippet
		for _, keyword := range keywords {
			if strings.Contains(snippet, keyword) {
				// 提取相关句子
				sentences := strings.Split(snippet, "。")
				for _, sentence := range sentences {
					if strings.Contains(sentence, keyword) && len(sentence) > 20 {
						update := simplifyText(sentence, 80)
						if update != "" {
							updates = append(updates, update)
							break
						}
					}
				}
				break
			}
		}
	}
	
	return updates
}

// extractMilitaryActions 提取军事行动
func extractMilitaryActions(results []SearchResult, limit int) []string {
	var actions []string
	keywords := []string{"袭击", "打击", "导弹", "无人机", "航母", "基地", "空袭"}

	for _, result := range results {
		if len(actions) >= limit {
			break
		}
		
		snippet := result.Snippet
		for _, keyword := range keywords {
			if strings.Contains(snippet, keyword) {
				// 提取相关句子
				sentences := strings.Split(snippet, "。")
				for _, sentence := range sentences {
					if strings.Contains(sentence, keyword) && len(sentence) > 20 {
						action := simplifyText(sentence, 80)
						if action != "" && !contains(actions, action) {
							actions = append(actions, action)
							break
						}
					}
				}
				break
			}
		}
	}
	
	return actions
}

// extractSources 提取信息来源
func extractSources(results []SearchResult) []string {
	var sources []string
	for _, result := range results {
		if len(sources) >= 3 {
			break
		}
		if result.Title != "" {
			sources = append(sources, fmt.Sprintf("%s - %s", 
				simplifyText(result.Title, 40), 
				extractDomain(result.URL)))
		}
	}
	return sources
}

// assessSituation 评估局势
func assessSituation(results []SearchResult) string {
	if len(results) == 0 {
		return "🟡 待观察 | ⚔️ 未知 | 🌍 待评估"
	}

	// 检查关键词
	criticalCount := 0
	criticalKeywords := []string{"死亡", "袭击", "爆炸", "打击", "导弹", "空袭"}
	
	for _, result := range results {
		text := result.Title + " " + result.Snippet
		for _, keyword := range criticalKeywords {
			if strings.Contains(text, keyword) {
				criticalCount++
				break
			}
		}
	}

	// 评估
	switch {
	case criticalCount >= 5:
		return "🔥🔥🔥🔥 高度危险 | ⚔️ 全面冲突 | 🌍 风险极高"
	case criticalCount >= 3:
		return "🔥🔥🔥 危险升级 | ⚔️ 持续冲突 | 🌍 风险高"
	case criticalCount >= 1:
		return "🔥🔥 紧张持续 | ⚔️ 有限冲突 | 🌍 风险中"
	default:
		return "🔥 相对稳定 | ⚔️ 低烈度 | 🌍 风险低"
	}
}

// saveBriefing 保存简报
func saveBriefing(briefing *Briefing, beijingTime time.Time) error {
	// 确保目录存在
	if err := os.MkdirAll(briefingDir, 0755); err != nil {
		return err
	}

	// 生成文件名
	filename := fmt.Sprintf("briefing-%s-%s.txt",
		beijingTime.Format("2006-01-02"),
		strings.Replace(briefing.TimeSlot, ":", "", -1))
	
	filepath := filepath.Join(briefingDir, filename)

	// 保存内容
	content := fmt.Sprintf("时间点: %s\n时段: %s\n生成时间: %s\n\n%s\n\n信息来源:\n",
		briefing.TimeSlot, briefing.Period, briefing.GeneratedAt, briefing.Content)
	
	for i, source := range briefing.Sources {
		content += fmt.Sprintf("%d. %s\n", i+1, source)
	}

	if err := ioutil.WriteFile(filepath, []byte(content), 0644); err != nil {
		return err
	}

	log.Printf("简报已保存: %s", filepath)
	return nil
}

// 辅助函数
func extractNumbers(text string) string {
	// 简化实现
	if strings.Contains(text, "死亡") {
		return "有伤亡报告"
	}
	return ""
}

func simplifyText(text string, maxLen int) string {
	// 清理文本
	text = strings.TrimSpace(text)
	text = strings.ReplaceAll(text, "\n", " ")
	text = strings.ReplaceAll(text, "#", "")
	
	// 截断
	if len(text) > maxLen {
		return text[:maxLen] + "..."
	}
	return text
}

func extractDomain(url string) string {
	// 简化实现
	if strings.Contains(url, "news.cn") {
		return "新华网"
	}
	if strings.Contains(url, "chinanews.com") {
		return "中新网"
	}
	if strings.Contains(url, "voachinese.com") {
		return "美国之音"
	}
	return "其他来源"
}

func contains(slice []string, item string) bool {
	for _, s := range slice {
		if s == item {
			return true
		}
	}
	return false
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}